### ISSUE TITLE
sudo does not clone the environment

### ISSUE THREAD

**Issue by sebthom** [chrisant996/sudo-windows#3]
It looks like the sudo command is not evaluating the PATH variable. E.g. I have vi/vim on PATH in my command window, but when I run e.g. `sudo vim <path_to_file>` I get:
```
'vim' is not recognized as an internal or external command,
operable program or batch file.
```

When running `where vim` I get:
```
C:\git\current\usr\bin\vim.exe
```

When I execute sudo using the full path of the exe it works as expected.

Interestingly, running e.g. `sudo cmd` or `sudo ping` works.

**Comment by chrisant996** [comment]
I cannot reproduce the problem.  Maybe the repro steps involve additional details that haven't been realized yet.

Maybe try `sudo set` and examine the environment for clues.

**Comment by sebthom** [comment]
When I execute `sudo set` the PATH variable does not show the paths that I added in the current command window session via `set "PATH=...;%PATH%"`. It seems to only show the persistent PATH variable content that one sets via `setx` or in the UI under `System Properties -> Environment Variables`.

I think under Linux one can preserve the current environment variables with -E / --preserve-env. Maybe something like this would be needed. But I am not sure if that is even possible in Windows. 

Also the only variable I really need preserved is the PATH variable. Something like this: https://www.baeldung.com/linux/sudo-command-not-found#2-fixing-the-error-permanently

**Comment by chrisant996** [comment]
Starting a new elevated process does not inherit from the previous process.  It's elevated.  That's a significant difference that can impact many environment variables.  While it's possible to override the new environment using variables from the old environment, that can cause malfunctions, and be misleading, and can potentially cause serious malfunctions.

**Comment by chrisant996** [comment]
I intentionally did not add a `-E` flag because it's so problematic, and I don't want to be yelled at about the problems when people shoot off their own feet by using it, so to speak.

**Comment by chrisant996** [comment]
gsudo has `--copyev` but says it's deprecated.  My guess is it's deprecated because it's problematic.

**Comment by chrisant996** [comment]
It's of course possible to create a `sudoers` kind of file, and enable various customizations in it, and define a location (or list of locations) where it can be located, etc.  I don't want to do quick partial hacks for degenerate custom cases, so it would have to be a "real" implementation, which begins to get complicated and expensive.  It's not on my radar to add.

Also, personally I use gsudo.  This sudo project was a learning prototype for me.  It is not intended as a fully supported project that will accumulate features and investment of time.

**Comment by sebthom** [comment]
I wasn't aware of gsudo, thanks for the hint. I just tried it but it can apparently not launch vim. I opened a bug report. 

Here are a lot of discussions https://unix.stackexchange.com/questions/83191/how-to-make-sudo-preserve-path regarding sudo on Linux, unfortunately, the easiest solution: `sudo $(which your_command)` does not translate to anything practical for the Windows Command Prompt since capturing command outputs is non-trivial in Batch and involves a lot of typing.

Do you think changing the sudo command to something like this would be feasable? 
1. check if the first argument is a valid relative or absolute path to an exe, if so, just execute it, otherwise
1. try to lookup the exe based on the PATH variable of the current command window session (e.g. using the where command) and replace the first argument with the absolute path resolved. then execute.

So basically do the `$(which your_command)` on behalf of the user inside the sudo process.

**Comment by chrisant996** [comment]
No. Not ok at all. Definitely breaks things. Will not consider partial quick hacks.

But you could consider enabling https://github.com/chrisant996/clink-gizmos/blob/main/command_substitution.lua and using that.

I'm categorically not investing further in this sudo project.

**Comment by sebthom** [comment]
Fair enough. I will look into the command substitution. Thank you.

If you don't plan to invest any more time into this project, it might be nice to archive this repo and point to the gsudo project.

**Comment by chrisant996** [comment]
The reason I didn't archive this is because you said this is the only sudo that meets your needs. 😜

Also, I'll fix core bugs. But I won't do hacks and I won't make significant extensions to it.

**Comment by sebthom** [comment]
> The reason I didn't archive this is because you said this is the only sudo that meets your needs. 😜

This is still the case and I am grateful you built it. 


> Also, I'll fix core bugs. But I won't do hacks and I won't make significant extensions to it. 

Gotcha.

**Comment by chrisant996** [comment]
Other options include 
1. make a luafunc: macro that runs `where (word under cursor)` and replaces the word under the cursor with that.
2. make doskey macros with full paths, and use a key binding to expand doskey macros before executing sudo command.
3. set path variable in registry rather than constructing it via scripts all the time.

I'd say option 3 is by far the best, and is the typical Best Practice. E.g. then you can e.g. run `vim` from the `Win-R` Run dialog, and etc.
